contracts/modules/maker/MakerV2Loan.sol

```diff
@@ -336,6 +336,7 @@ abstract contract MakerV2Loan is MakerV2Base {
     )
         external
         onlyModule(_wallet)
+        onlyNewVersion
```

Comment:
In our Maker integration vaults are technically speaking owned by the `MakerV2Manager` module (but controlled by the users). When we upgrade the `MakerV2Manager` the vaults need to be transferred to the new version of the module. That's what the `giveVault` method does, and that's why it could only be called by an authorised module, i.e. a potential new version of the `MakerV2Manager`. However, following this PR our model has changed and the  `RelayerModule` can now call all the other modules. Making sure that `giveVault` is only called by an authorised module is no longer sufficient since it does not prevent the `RelayerModule` from calling it (which would transfer the vault to the `RelayerModule` and the vault would be stuck... no reason for a user to do that on one of its vault but it should nonetheless be prevented).

That's why I have added the `onlyNewVersion` modifier which implements a callback to the module calling the `giveVault` method. If the new version of `MakerV2Manager` responds to the callback it will be the only module with the right to transfer the vaults. Should a user try to call `giveVault` via the `RelayerModule` and it will revert as expected.